utils/addCENNZnetToMetaMask.ts

```diff
@@ -0,0 +1,36 @@
+import { MetaMaskInpageProvider } from "@metamask/providers";
+import { CENNZ_METAMASK_NETWORK, CENNZ_NETWORK } from "@/constants";
+
+export default async function addCENNZnetToMetaMask(
+	extension: MetaMaskInpageProvider
+): Promise<void> {
+	const ethChainId = await extension.request({ method: "eth_chainId" });
+
+	if (ethChainId === CENNZ_METAMASK_NETWORK.chainId) return;
+
+	try {
+		await extension.request({
+			method: "wallet_switchEthereumChain",
+			params: [{ chainId: CENNZ_METAMASK_NETWORK.chainId }],
+		});
+	} catch (error) {
+		if (error.code === 4902) {
```

Comment:
use the updated version in `app-faucet` with return early and re-throw non `4902` error code